rpcs3/Emu/Cell/Modules/cellAdec.h

```diff
@@ -370,16 +317,20 @@ enum CellAdecMsgType : s32
 
 using CellAdecCbMsg = s32(u32 handle, CellAdecMsgType msgType, s32 msgData, u32 cbArg);
 
-struct CellAdecCb
+// Used for internal callbacks as well
+template <typename F>
+struct AdecCb
 {
-	vm::bptr<CellAdecCbMsg> cbFunc;
-	be_t<u32> cbArg;
+	vm::bptr<F> cbFunc;
+	vm::bptr<void> cbArg;
 };
 
+typedef AdecCb<CellAdecCbMsg> CellAdecCb;
```

Comment:
You can use `using` here as well, or is there some issue?

Reply:
No idea why I used `typedef` there xD